src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp

```diff
@@ -376,6 +376,27 @@ std::string LevelZeroCompilerInDriver<TableExtension>::serializeConfig(
     const Config& config,
     ze_graph_compiler_version_info_t& compilerVersion) const {
     std::string content = config.toString();
+    _logger.debug("Original content of config: %s", content.c_str());
+
+    // Remove optimization-level and performance-hint-override for old driver which not support them
+    if ((compilerVersion.major < 5) || (compilerVersion.major == 5 && compilerVersion.minor < 7)) {
+        std::ostringstream optLevelStr;
+        optLevelStr << "optimization-level" << KEY_VALUE_SEPARATOR << "\\d+";
+        std::ostringstream perfHintStr;
+        perfHintStr << "performance-hint-override" << KEY_VALUE_SEPARATOR << "\\S+";
+        _logger.warning(
```

Comment:
Should not we first check if `ptimization-level`/`performance-hint-override` are present?

```
auto areOptionsSupported = (compilerVersion.major > 5) || (compilerVersion.major == 5 && compilerVersion.minor > 7);
auto contaisOptions = content.contains(optLevelStr) || content.contains(perfHintStr); // pseudo-code

if (!areOptionsSupported  && contaisOptions) { 
// write message
// remove options
}
```

Reply:
Added more strict check now, please help to review again